### PR TITLE
GQI - Add IGQIOnInit and IGQIOnDestory methods to custom operators

### PR DESCRIPTION
**Pull request by WaHaeck**

### CODE REVIEW

**Review by skyline-rge — CHANGES_REQUESTED** [review]

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Ad_hoc_data/Ad_hoc_Building_blocks.md [hunk 1]
```diff
@@ -14,33 +14,17 @@ All other interfaces add additional functionality.
...
   > This is the only required interface. See also: [Configuring an ad hoc data source in a query](xref:Configuring_an_ad_hoc_data_source_in_a_query)
 
-- [IGQIInputArguments](xref:GQI_IGQIInputArguments)
+- [IGQIInputArguments](xref:GQI_IGQIInputArguments): Retrieve input from the user through input arguments.
 
-- [IGQIOnDestroy](xref:GQI_IGQIOnDestroy)
+- [IGQIOnDestroy](xref:GQI_IGQIOnDestroy): Can be implemented to clean up resources after it has been used.
 
-- [IGQIOnInit](xref:GQI_IGQIOnInit)
+- [IGQIOnInit](xref:GQI_IGQIOnInit): Can be implemented to get notified when the custom operator is created.
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Ad_hoc_data/Ad_hoc_Building_blocks.md:26]: Would write something like this instead:
Provides a way to initialize the data source with access to dependencies like e.g. the DMS

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Custom_Operator/CO_Building_blocks.md [hunk 2]
```diff
@@ -22,6 +22,6 @@ The available interfaces are:
 
 - [IGQIOnDestroy](xref:GQI_IGQIOnDestroy): Can be implemented to clean up resources after it has been used. (available from DataMiner 10.4.5/10.5.0 onwards<!-- RN 38959 -->)
 
-- [IGQIOnInit](xref:GQI_IGQIOnInit): Can be implemented to get notified when the custom operator is created. (available from DataMiner 10.4.5/10.5.0 onwards<!-- RN 38959 -->)
+- [IGQIOnInit](xref:GQI_IGQIOnInit): Provides a way to initialize the data source with access to dependencies like the DMS. (available from DataMiner 10.4.5/10.5.0 onwards<!-- RN 38959 -->)
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Custom_Operator/CO_Building_blocks.md:25]: custom operator

**Review by skyline-rge — CHANGES_REQUESTED** [review]

**Review by skyline-rge — APPROVED** [review]

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQIOptimizableOperator.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    ConfigureMyCustomLogicFor(nextOperator);
+
+    // Return our current node that is now also responsible for the next operation
+    return currentNode;
+}
+```
+
+##### Unconditionally forward
+
+To forward any operator, i.e. to give it more priority than your own custom operator, you can call the `Forward(IGQICoreOperator)` on the current `IGQIOperatorNode`.
```
Comment (MariekeGO) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQIOptimizableOperator.md:76]: I'd prefer to rephrase this to avoid the "your own", as this sounds a bit odd here. We can't assume that the operator was made by whoever is reading this, so it's better to make this less personal. Would this be OK?
```suggestion
To forward any operator, i.e. to give it more priority than another custom operator, you can call `Forward(IGQICoreOperator)` on the current `IGQIOperatorNode`.
```

**Review by MariekeGO — APPROVED** [review]
